CHAID/CHAID.py

```diff
@@ -389,12 +389,12 @@ def generate_best_split(self, ind, dep):
 
                 choice, chi, highest_p_split = max(sub_data, key=lambda x: (x[2], x[1]))
 
-                if size == 1 or highest_p_split < self.alpha_merge:
+                if highest_p_split < self.alpha_merge:
```

Comment:
`size` is (independent variables CHOOSE two) so size = 1 iff there is only one possible grouping. I think it was intended as an escape the loop early case.